src/App.js

```
@@ -582,159 +632,207 @@ class App extends Component {
+                this.setState({ syncStatus });
+                this.pushToLocalStorage(
+                  binding.id,
+                  this.state.url,
+                  binding.id.replace('dw::', ''),
+                  binding.rangeAddress,
+                  currentSelectedRange.worksheet.id,
+                  new Date()
+                );
+                this.setState({ url: '', selectSheet: false });
+                resolve();
+              })
+              .catch((error) => {
+                this.setState({ error });
+                this.setState({ syncing: false });
+                reject();
               });
-            }).then(() => {
-              const syncStatus = this.state.syncStatus;
-              syncStatus[binding.id].synced = true;
-              syncStatus[binding.id].changes = 0;
-              syncStatus[binding.id].lastSync = new Date();
-              this.office.setSyncStatus(syncStatus);
-              this.setState({ syncStatus });
-              this.pushToLocalStorage(binding.id, this.state.url, binding.id.replace('dw::', ''), binding.rangeAddress, currentSelectedRange.worksheet.id, new Date())
-              this.setState({url: '', selectSheet: false})
-              resolve();
-            }).catch((error) => {
-              this.setState({error});
-              this.setState({syncing: false});
-              reject();
-            });
           });
           promises.push(promise);
         });
-        Promise.all(promises).then(() => {
-          this.setState({syncing: false});
-          resolve();
-        }).catch((error) => {
-          this.setState({syncing: false});
-          reject();
-        });
+        Promise.all(promises)
+          .then(() => {
+            this.setState({ syncing: false });
+            resolve();
+          })
+          .catch((error) => {
+            this.setState({ syncing: false });
+            reject();
+          });
+      });
+    } catch (updateBindingError) {
```

Catch all block can be replaced by global error boundary.